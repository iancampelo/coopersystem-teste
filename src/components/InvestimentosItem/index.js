import React from 'react'
import {
    StyleSheet,
    Text,
    TouchableHighlight,
    View,
} from 'react-native'
import { normalizeMoneyString } from '../../util/functions'

const indicadorCarenciaTrue = 'S'
const indicadorCarenciaFalse = 'N'

const InvestimentosItem = (props) => {
  const {investimento, navigation} = props
  const carencia = investimento.indicadorCarencia === indicadorCarenciaTrue

  return (
    <TouchableHighlight
      style={[styles.investimentoItem, carencia ? styles.itemListDisabled : styles.itemList]}
      disabled={carencia}
      onPress={() =>
        navigation.navigate(
          'Resgate', {
            valorInvestimentos: investimento.saldoTotalDisponivel,
            acoes: investimento.acoes,
            nome: investimento.nome
          }
        )
      }>
      <View style={styles.itemContainer}>
        <Text style={styles.itemNome}>
          {investimento.nome} - {investimento.objetivo}
        </Text>
        <Text style={styles.itemValor}>
          {normalizeMoneyString(investimento.saldoTotalDisponivel)}
        </Text>
      </View>
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  investimentoItem: {
    margin: 10,
    height: 60,

  },
  itemList: {
    backgroundColor: "white",
  },
  itemListDisabled: {
    backgroundColor: "gray",
  },
  itemContainer: {
    flex: 1,
    alignItems: "center",
    height: 60,
    padding: 15,
  },
  itemNome: {
    fontSize: 14,
    fontWeight: "bold",
  },
});

export default InvestimentosItem
