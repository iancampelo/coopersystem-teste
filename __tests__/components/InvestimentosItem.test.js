import 'react-native'
import React from 'react'
import InvestimentosItem from '../../src/components/InvestimentosItem'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'
jest.useFakeTimers()

const mockAcoes = {
  nome: 'INVESTIMENTO I',
  objetivo: 'Minha aposentadoria',
  saldoTotalDisponivel: 39321.29,
  indicadorCarencia: 'N',
  acoes: [
    {
      id: '1',
      nome: 'BBAS3',
      percentual: 28.1,
    },
    {
      id: '2',
      nome: 'VALE3',
      percentual: 20.71,
    },
    {
      id: '3',
      nome: 'PETR4',
      percentual: 21.63,
    },
    {
      id: '4',
      nome: 'CMIG3',
      percentual: 12.41,
    },
    {
      id: '5',
      nome: 'OIBR3',
      percentual: 17.15,
    },
  ],
};

it('renders correctly InvestimentosItem', () => {
  const tree = renderer.create(
    <InvestimentosItem
      navigation={jest.fn()}
      investimento={mockAcoes}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
