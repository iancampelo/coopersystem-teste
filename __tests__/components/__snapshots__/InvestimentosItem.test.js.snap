// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly InvestimentosItem 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 60,
        "margin": 10,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 60,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
        }
      }
    >
      INVESTIMENTO I
       - 
      Minha aposentadoria
    </Text>
    <Text>
      R$39,321.29
    </Text>
  </View>
</View>
`;
